cli/module_graph2.rs

```diff
@@ -566,24 +668,150 @@ impl Graph2 {
     self.modules.contains_key(s)
   }
 
+  /// Type check the module graph, corresponding to the options provided.
+  pub fn check(
+    self,
+    options: CheckOptions,
+  ) -> Result<(Stats, Diagnostics, Option<IgnoredCompilerOptions>), AnyError>
+  {
+    debug!(
+      "check: emit={} lib={:?} maybe_config_path={:?}, reload={}",
+      options.emit, options.lib, options.maybe_config_path, options.reload
+    );
+    let mut config = TsConfig::new(json!({
+      "allowJs": true,
+      // TODO(@kitsonk) is this really needed?
+      "esModuleInterop": true,
+      // Enabled by default to align to transpile/swc defaults
+      "experimentalDecorators": true,
+      "incremental": true,
+      "isolatedModules": true,
```

Comment:
Note, this is a breaking change that was planned for 1.5... See: #7326

Reply:
Thanks for pointing this out - Can we do this in a separate PR?

Reply:
Ok, taken out and will follow it up in a seperate PR.

Reply:
I'll reopen #8050.